src/gaugeProject.ts

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+import { existsSync, readFileSync } from 'fs';
+import { GAUGE_MANIFEST_FILE, MAVEN_POM } from './constants';
+import { join, parse, relative, isAbsolute } from 'path';
+import { CLI } from './cli';
+
+export class GaugeProject {
+    private readonly _projectRoot: string;
+    private readonly _isGaugeProject: boolean;
+    private readonly _language: any;
+    private readonly _plugins: any;
+
+    public constructor(projectRoot: string, manifest: any) {
+        this._projectRoot = projectRoot;
+        this._isGaugeProject = manifest != null;
+        if (this._isGaugeProject) {
+            this._language = manifest.Language;
+            this._plugins = manifest.Plugins;
+        }
+    }
+
+    public getExecutionCommand(cli: CLI): string {
+        if (this.isMavenProject()) return cli.mavenCommand();
+        return cli.gaugeCommand();
+    }
+
+    public isMavenProject() {
```

Comment:
perhaps `MavenProject` can inherit from`GaugeProject` and a factory to return the right project type?